File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/CallbackQueryFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram;

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter;
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface CallbackQueryFilter {

    MessageFilter message() default @MessageFilter(status = AnnotationState.OFF);

    String[] validator() default {};

    StringFilter inlineMessageId() default @StringFilter(status = AnnotationState.OFF);

    StringFilter chatInstance() default @StringFilter(status = AnnotationState.OFF);

    StringFilter data() default @StringFilter(status = AnnotationState.OFF);

    StringFilter gameShortName() default @StringFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/CallbackQueryFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class CallbackQueryFilter(val message: MessageFilter = MessageFilter(status = AnnotationState.OFF), val validator: Array<String> = [], val inlineMessageId: StringFilter = StringFilter(status = AnnotationState.OFF), val chatInstance: StringFilter = StringFilter(status = AnnotationState.OFF), val data: StringFilter = StringFilter(status = AnnotationState.OFF), val gameShortName: StringFilter = StringFilter(status = AnnotationState.OFF))
